llvm/utils/emacs/llvm-mode.el

```diff
@@ -61,6 +61,8 @@
    `(,(concat "\\<" llvm-mode-primitive-type-regexp "\\>") . font-lock-type-face)
    ;; Integer literals
    '("\\b[-]?[0-9]+\\b" . font-lock-preprocessor-face)
+   ;; Values that can appear in a vec
+   '("\\b\\(true\\|false\\|null\\|undef\\|poison\\|none\\)\\b" . font-lock-keyword-face)
```

Comment:
I wonder if the `'words` paren argument to regexp-opt would also work? https://www.gnu.org/software/emacs/manual/html_node/elisp/Regexp-Functions.html#index-regexp_002dopt

```elisp
`(,(regexp-opt '("true" "false" "null" "undef" "none" "poison") 'words) . font-lock-keyword-face
```

Reply:
Let me check.

BTW, first I tried `(mapconcat 'identity '("true" "false" "null" "undef" "none" "poison") "//|")`  for the list but that didn't work, any idea why?

Reply:
doesn't seem to work.

Reply:
> BTW, first I tried `(mapconcat 'identity '("true" "false" "null" "undef" "none" "poison") "//|")` for the list but that didn't work, any idea why?

Did you mean `\\|`?

Reply:
> doesn't seem to work.

Ah well, was worth a shot

Reply:
> > BTW, first I tried `(mapconcat 'identity '("true" "false" "null" "undef" "none" "poison") "//|")` for the list but that didn't work, any idea why?
> 
> Did you mean `\\|`?

Yeah

Reply:
Full code:
```
'((concat "\\b\\(" (mapconcat 'identity '("true" "false" "null" "undef" "none" "poison") "\\|") "\\)\\b") . font-lock-keyword-face)
```

Reply:
Also LGTM!

Reply:
Err that doesn't work, any idea why?

Reply:
No idea. Evaluating your snippet gives me `"\\b\\(true\\|false\\|null\\|undef\\|none\\|poison\\)\\b"` which is the same as whats in this PR, but none and poison are swapped around?

Reply:
Yeah, I don't know either, cest la vie, ill merge this as is.